lib/active_merchant/billing/gateways/latitude19.rb

```diff
@@ -0,0 +1,416 @@
+          end
+        end
+      end
+
+      def authorize(amount, payment_method, options={})
+        if payment_method.is_a?(String)
+          auth_or_sale("auth", payment_method, amount, nil, options)
+        else
+          MultiResponse.run() do |r|
+            r.process { get_session(options) }
+            r.process { get_token(r.authorization, payment_method, options) }
+            r.process { auth_or_sale("auth", r.authorization, amount, payment_method, options) }
+          end
+        end
+      end
+
+      def capture(amount, authorization, options={})
+        post = {}
+        post[:method] = "deposit"
+        add_request_id(post)
+
+        params = {}
+
+        _, params[:pgwTID] = split_authorization(authorization)
+
+        add_invoice(params, amount, options)
+        add_credentials(params, post[:method])
+
+        post[:params] = [params]
+        commit("v1/", post)
+      end
+
+      def void(authorization, options={})
+        method, pgwTID = split_authorization(authorization)
+        case method
+        when "auth"
+          reverse_or_void("reversal", pgwTID, options)
+        when "deposit", "sale"
+          reverse_or_void("void", pgwTID, options)
+        else
+          message = "Unsupported operation: successful Purchase, Authorize and unsettled Capture transactions can only be voided."
+          return Response.new(false, message)
+        end
+      end
+
+      def credit(amount, payment_method, options={})
+        if payment_method.is_a?(String)
+          refundWithCard(payment_method, amount, nil, options)
+        else
+          MultiResponse.run() do |r|
+            r.process { get_session(options) }
+            r.process { get_token(r.authorization, payment_method, options) }
+            r.process { refundWithCard(r.authorization, amount, payment_method, options) }
+          end
+        end
+      end
+
+      def verify(payment_method, options={}, action=nil)
+        if payment_method.is_a?(String)
+          verifyOnly(action, payment_method, nil, options)
+        else
+          MultiResponse.run() do |r|
+            r.process { get_session(options) }
+            r.process { get_token(r.authorization, payment_method, options) }
+            r.process { verifyOnly(action, r.authorization, payment_method, options) }
+          end
+        end
+      end
+
+      def store(payment_method, options={})
+        verify(payment_method, options, "store")
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((\"cardNumber\\\":\\\")\d+), '\1[FILTERED]').
+          gsub(%r((\"cvv\\\":\\\")\d+), '\1[FILTERED]')
+      end
+
+
+      private
+
+      def add_request_id(post)
+        post[:id] = SecureRandom.hex(16)
+      end
+
+      def add_timestamp()
+        Time.now.getutc.strftime("%Y%m%d%H%M%S")
+      end
+
+      def add_hmac(params, method)
+        if method == "getSession"
```

Comment:
I believe `getSession` is an endpoint we were invoking only for testing? If so, we should remove references to it from the adapter to keep it clean and debug-free.